### PR TITLE
fix(sdk): fix typescript type references to "embedding-sdk" not resolving

### LINKED ISSUE #47627: If we export something that relies on an import from "embedding-sdk" the users of the sdk will see it types as any
Thread 1: https://metaboat.slack.com/archives/C063Q3F1HPF/p1723023805043939

Thread 2: https://metaboat.slack.com/archives/C063Q3F1HPF/p1725464383559709

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/47627

### Description

Fixes the fixup script so it now replaces both top-level imports e.g. `import whatever from "embedding-sdk"` and imports with slash, e.g. `import whatever from "embedding-sdk/foo/bar"`

### How to verify

- `yarn build-embedding-sdk`
   - This actually runs `yarn embedding-sdk:fixup-types-imports` under the hood
- Go to `resources/embedding-sdk/dist/enterprise/frontend/src/embedding-sdk`
   - Open full text search in your IDE of choice
   - Search for `from "embedding-sdk"` - you should notice they are all prefixed by `from "@metabase/embedding-sdk-react/dist/...`
- Now go to your React project, and run this command `rm -rf node_modules/@metabase && yarn add file:../metabase/resources/embedding-sdk && rm -rf node_modules/.vite`
- Look at the types for `CreateQuestion` and `InteractiveQuestion`, they should resolve fine now.
   - Notice that there is no more reference to "embedding-sdk" there

### Demo

Before

image

After

CleanShot 2567-09-05 at 16 07 42@2x

### CODE REVIEW

File: bin/embedding-sdk/fixup-types-after-compilation.js [hunk 1]
```diff
@@ -14,14 +14,12 @@ const SDK_PACKAGE_NAME = "@metabase/embedding-sdk-react";
 
 // this map should be synced with "tsconfig.sdk.json"
 const REPLACES_MAP = {
-  "metabase-enterprise/": `${SDK_PACKAGE_NAME}/dist/enterprise/frontend/src/metabase-enterprise/`,
-  "metabase-lib/": `${SDK_PACKAGE_NAME}/dist/frontend/src/metabase-lib/`,
+  "metabase-enterprise": `${SDK_PACKAGE_NAME}/dist/enterprise/frontend/src/metabase-enterprise`,
   "metabase-lib": `${SDK_PACKAGE_NAME}/dist/frontend/src/metabase-lib`,
-  "metabase-shared/": `${SDK_PACKAGE_NAME}/dist/frontend/src/metabase-shared/`,
```
Comment (heypoom) [inline bin/embedding-sdk/fixup-types-after-compilation.js:20]: We no longer references `metabase-shared`

File: bin/embedding-sdk/fixup-types-after-compilation.js [hunk 2]
```diff
@@ -14,14 +14,12 @@ const SDK_PACKAGE_NAME = "@metabase/embedding-sdk-react";
...
-  "metabase-types/": `${SDK_PACKAGE_NAME}/dist/frontend/src/metabase-types/`,
-  "metabase/": `${SDK_PACKAGE_NAME}/dist/frontend/src/metabase/`,
-  "embedding-sdk/": `${SDK_PACKAGE_NAME}/dist/enterprise/frontend/src/embedding-sdk/`,
-  "cljs/": `${SDK_PACKAGE_NAME}/dist/target/cljs_release/`,
+  "metabase-types": `${SDK_PACKAGE_NAME}/dist/frontend/src/metabase-types`,
+  "metabase": `${SDK_PACKAGE_NAME}/dist/frontend/src/metabase`,
+  "embedding-sdk": `${SDK_PACKAGE_NAME}/dist/enterprise/frontend/src/embedding-sdk`,
+  "cljs": `${SDK_PACKAGE_NAME}/dist/target/cljs_release`,
```
Comment (heypoom) [inline bin/embedding-sdk/fixup-types-after-compilation.js:22]: Removed the slash here so we can only have a single variant. We'll append it below.

File: bin/embedding-sdk/fixup-types-after-compilation.js [hunk 3]
```diff
@@ -49,7 +47,10 @@ const replaceAliasedImports = filePath => {
 
   Object.entries(REPLACES_MAP).forEach(([alias, replacement]) => {
     fileContent = fileContent
-      .replaceAll(`from "${alias}`, `from "${replacement}`)
+      // replaces "metabase-lib/foo" with "<sdk>/metabase-lib/foo"
+      .replaceAll(`from "${alias}/`, `from "${replacement}/`)
+      // replaces "metabase-lib" with "<sdk>/metabase-lib"
+      .replaceAll(`from "${alias}"`, `from "${replacement}"`)
```
Comment (heypoom) [inline bin/embedding-sdk/fixup-types-after-compilation.js:53]: Handles both variations of import.

File: bin/embedding-sdk/fixup-types-after-compilation.js (hunk 1 above)
Comment (npretto) [reply bin/embedding-sdk/fixup-types-after-compilation.js:20]: Are we sure there may not be a change in the main app that could cause an import from there?

**Review by npretto — APPROVED** [review]
left a question but otherwise looks good

File: bin/embedding-sdk/fixup-types-after-compilation.js (hunk 1 above)
Comment (heypoom) [reply bin/embedding-sdk/fixup-types-after-compilation.js:20]: @npretto I think it is actually not referenced by any JS/TS file. From full-text search, the only reference is inside a comment (out-of-date), a webpack file, and from pulse:

```
(def ^:private js-file-path "frontend_shared/color_selector.js")
```

CleanShot 2567-09-05 at 16 24 29@2x

**Review by WiNloSt — APPROVED** [review]
Tested, this works great!

**Pull request merged**